File: src/main/java/clients/UDP_FTP/FTPClient.java
package clients.UDP_FTP;

import java.io.FileOutputStream;
import java.io.IOException;
import org.apache.commons.net.ftp.FTP;

public class FTPClient {

    public static void main(String[] args) {
        String server = "109.167.241.225";
        int port = 21;
        String user = "Student";
        String pass = "FksG5$%^rgtdSDFH";
        String remoteFile = "UDP log.txt";

        org.apache.commons.net.ftp.FTPClient ftpClient = new org.apache.commons.net.ftp.FTPClient();
        try {
            ftpClient.connect(server, port);
            ftpClient.login(user, pass);
            ftpClient.enterLocalPassiveMode();
            ftpClient.setFileType(FTP.BINARY_FILE_TYPE);

            FileOutputStream outputStream = new FileOutputStream(remoteFile);
            boolean success = ftpClient.retrieveFile(remoteFile, outputStream);
            outputStream.close();

            if (success)
                System.out.println("File downloaded successfully.");
             else
                System.out.println("Cant download file.");

        } catch (IOException ex) {
            System.out.println("Ошибка: " + ex.getMessage());
            ex.printStackTrace();
        } finally {
            try {
                if (ftpClient.isConnected()) {
                    ftpClient.logout();
                    ftpClient.disconnect();
                }
            } catch (IOException ex) {
                ex.printStackTrace();
            }
        }

    }
}

File: src/main/java/clients/UDP_FTP/ftp_client.go
package UDP_FTP

import (
	"fmt"
	"github.com/jlaffaye/ftp"
	"io"
	"os"
	"strings"
)

func main() {
	server := "109.167.241.225"
	port := 21
	user := "Student"
	pass := "FksG5$%^rgtdSDFH"
	remoteFile := "UDP log.txt"

	// Подключение к FTP серверу
	ftpClient, err := ftp.Dial(fmt.Sprintf("%s:%d", server, port))
	if err != nil {
		fmt.Printf("Ошибка при подключении к FTP серверу: %v\n", err)
		return
	}
	defer ftpClient.Quit()

	err = ftpClient.Login(user, pass)
	if err != nil {
		fmt.Printf("Ошибка при аутентификации: %v\n", err)
		return
	}

	// Получение файла с FTP сервера
	file, err := os.Create(remoteFile)
	if err != nil {
		fmt.Printf("Ошибка при создании файла: %v\n", err)
		return
	}
	defer file.Close()

	err = ftpClient.Retrieve(remoteFile, file)
	if err != nil {
		fmt.Printf("Ошибка при получении файла: %v\n", err)
		return
	}

	fmt.Println("Файл успешно скачан.")
}
